### ISSUE TITLE
Accessing to the raw content of astropy.io.votable.tree.Resource(s)

### ISSUE THREAD

**Issue by lmichel** [astropy/astropy#12153]
### Description
In the frame of the VO data modeling efforts, we have been planning to annotate VOTable resources with data model meta-data.

These annotations are XML pieces matching a specific schema independent of the VOTable schema.  An annotated VOTable (work in progress) can be seen [here](https://github.com/ivoa-std/ModelInstanceInVot/blob/votable-schema/schema/validation/snippets/test_rich_instance_1_ok.xml) (branch `votable-schema`)

Annotations are not intended to be consumed by Astropy but rather by some affiliated  package. Anyway,  `astropy.io.votable` must be enabled to provide access to that XML for any end-client. 

In order to avoid to modify the current VOTable schema, these annotations will be enclosed in a Resource of type `meta` with a specific name space:

```xml
 <VOTable>
     <RESOURCE type=result>
         <RESOURCE type=meta>
             <dm-mapping:VODML ... My DM stuff ....>
        </RESOURCE>
       <TABLE/>       
       <TABLE/>
     </RESOURCE>
</VOTABLE>
```
This approach has been validated with regular XML parsers and validators and we need Astropy to support this new VOTable feature.
For this,  the`astropy.io.votable` API must be able to extract the content of such resources. This is apparently not possible yet.

The new endpoint should look like:
```python
votable = parse(votable_path)
   for resource in votable.resources:
       if resource.type == "results":
           for sub_resource in resource.resources:
               if sub_resource.type == "meta":
                   print("Here is my DM stuff")
                   annotation_content = sub_resource.content()
```

The resource content might be delivered as a DOM or/and as a string. The latest  would allow final clients to run their own XML processors.

**Comment by pllim** [comment]
@tomdonaldson , I remember you said on Slack that you are interested in tackling this, is that still true? 😸

**Comment by lmichel** [comment]
I've been prototyping this feature on my [fork](https://github.com/lmichel/astropy) with a read/write working [example](https://github.com/lmichel/astropy/tree/VODML_readout/examples/modelmapping).

Our guideline (@tomdonaldson and myself) is that the piece of XML we need to work with (the model mapping block) must be extracted by the VOTable parser (tree.py) in a safe way, but it is not meant to by used by any Astropy function, VO data models being quite far from the Astropy core business. AstroPy is just an agnostic provider in this case.

I enabled `tree.py` to extract all XML elements contained in the `Resource@type=meta`; these elements are serialized as a string that can be provided with a specific getter. 
- There no syntax validation but the element names. 
- This is consistent with  the requirement that errors in the mapping must not break the VOTable validity.
I've also implemented the reverse operation for writing model mapping blocks in VOtables.

I'm fully aware that this post must be opaque to people not connected with the VO DM discussion and I'm ready to answer any question.
However, I would appreciate  if someone could tell me whether my way to proceed is correct or no.
